src/main/java/su/litvak/chromecast/api/v2/ChromeCasts.java

```diff
@@ -21,61 +21,58 @@
 import java.io.IOException;
 import java.net.InetAddress;
 import java.util.ArrayList;
+import java.util.Collections;
 import java.util.List;
 
 /**
  * Utility class that discovers ChromeCast devices and holds references to all of them.
  */
-public class ChromeCasts extends ArrayList<ChromeCast> implements ServiceListener {
+public class ChromeCasts {
     private final static ChromeCasts INSTANCE = new ChromeCasts();
+    private final MyServiceListener listener = new MyServiceListener();
 
     private JmDNS mDNS;
 
-    private List<ChromeCastsListener> listeners = new ArrayList<ChromeCastsListener>();
-
+    private final List<ChromeCastsListener> listeners = new ArrayList<ChromeCastsListener>();
+    private final List<ChromeCast> chromeCasts;
+    
     private ChromeCasts() {
+        chromeCasts = Collections.synchronizedList(new ArrayList<ChromeCast>());
```

Comment:
Why have you moved field initialization from declaration to constructor? I think we should leave it in declaration for uniformity.